cmd/bootnode/main.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/athanorlabs/atomic-swap/rpc"
+
+	"github.com/hashicorp/go-multierror"
+	logging "github.com/ipfs/go-log"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	defaultLibp2pPort = 9900
+	defaultRPCPort    = common.DefaultSwapdPort
+
+	flagDataDir    = "data-dir"
+	flagLibp2pKey  = "libp2p-key"
+	flagLibp2pPort = "libp2p-port"
+	flagBootnodes  = "bootnodes"
+	flagRPCPort    = "rpc-port"
+	flagEnv        = "env"
+)
+
+var log = logging.Logger("cmd")
+
+func cliApp() *cli.App {
+	return &cli.App{
+		Name:                 "bootnode",
+		Usage:                "A bootnode for the atomic swap p2p network.",
+		Version:              cliutil.GetVersion(),
+		Action:               runBootnode,
+		EnableBashCompletion: true,
+		Suggest:              true,
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  flagDataDir,
+				Usage: "Path to store swap artifacts",
+				Value: "{HOME}/.atomicswap/{ENV}", // For --help only, actual default replaces variables
+			},
+			&cli.StringFlag{
+				Name:  flagLibp2pKey,
+				Usage: "libp2p private key",
+				Value: fmt.Sprintf("{DATA_DIR}/%s", common.DefaultLibp2pKeyFileName),
+			},
+			&cli.UintFlag{
+				Name:  flagLibp2pPort,
+				Usage: "libp2p port to listen on",
+				Value: defaultLibp2pPort,
+			},
+			&cli.StringSliceFlag{
+				Name:    flagBootnodes,
+				Aliases: []string{"bn"},
+				Usage:   "libp2p bootnode, comma separated if passing multiple to a single flag",
+				EnvVars: []string{"SWAPD_BOOTNODES"},
+			},
+			&cli.UintFlag{
+				Name:  flagRPCPort,
+				Usage: "Port for the bootnode RPC server to run on",
+				Value: defaultRPCPort,
+			},
+			&cli.StringFlag{
+				Name:  flagEnv,
+				Usage: "Environment to use: one of mainnet, stagenet, or dev",
+				Value: "dev",
+			},
+			&cli.StringFlag{
+				Name:  cliutil.FlagLogLevel,
+				Usage: "Set log level: one of [error|warn|info|debug]",
+				Value: "info",
+			},
+		},
+	}
+}
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go cliutil.SignalHandler(ctx, cancel, log)
+
+	err := cliApp().RunContext(ctx, os.Args)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func runBootnode(c *cli.Context) error {
+	// Fail if any non-flag arguments were passed
+	if c.Args().Present() {
+		return fmt.Errorf("unknown command %q", c.Args().First())
+	}
+
+	if err := cliutil.SetLogLevelsFromContext(c); err != nil {
+		return err
+	}
+
+	config, err := getEnvConfig(c)
+	if err != nil {
+		return err
+	}
+
+	libp2pKeyFile := config.LibP2PKeyFile()
+	if c.IsSet(flagLibp2pKey) {
+		libp2pKeyFile = c.String(flagLibp2pKey)
+		if libp2pKeyFile == "" {
+			return errFlagValueEmpty(flagLibp2pKey)
+		}
+	}
+
+	if libp2pKeyFile == "" {
+		libp2pKeyFile = path.Join(config.DataDir, common.DefaultLibp2pKeyFileName)
+	}
+
+	libp2pPort := uint16(c.Uint(flagLibp2pPort))
+
+	hostListenIP := "0.0.0.0"
+	if config.Env == common.Development {
+		hostListenIP = "127.0.0.1"
+	}
+
+	host, err := net.NewHost(&net.Config{
```

Comment:
I know it will be a super tiny package, but can we separate out all the code that is not dealing with CLI or environment variables and push it into a `bootnode` package. I will definitely be writing unit tests that use the dedicated bootnode.